### PR TITLE
Add "last logged in" field.

### LINKED ISSUE #543: Store Last Login Date in Northstar and Expose via API
Analagus to Phoenix `last_logged_in` field, this piece of data will help track user activity and MAU. 

Ideally would backfill in dates from Phoenix.

### PR DESCRIPTION
**Pull request by DFurnes**
#### What's this PR do?
This pull request adds a `last_authenticated_at` field to the user model, and updates it whenever the user successfully authenticates via the Registrar. It also fires a StatHat "user authentication" stat counter because heck why not. 🙆‍♂️

Closes #543.

#### How should this be reviewed?
Okay okay, I know it's lot of _files_ but it's not that many _lines_! 🙃

#### Checklist
- [ ] Documentation added for changed endpoints.
- [ ] Tests added for new features/bug fixes.
- [ ] Post a message in #api if this includes something that causes a rebuild!

### CODE REVIEW

**Comment by DFurnes** [comment]
[(3) As a data analyst, I want Northstar to collect and add fields for `last_logged_in`](https://trello.com/c/ZG5aeGAZ/278-3-as-a-data-analyst-i-want-northstar-to-collect-and-add-fields-for-last-logged-in)

**Comment by DFurnes** [comment]
@sheyd I used `last_authenticated_at` as the field name for consistency with the other timestamps. Let me know if that sounds okay to you!

File: app/Listeners/SuccessfulAuthentication.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    /**
+     * Handle the event.
+     *
+     * @param Login $event
+     * @return void
+     * @internal param User $user
+     */
+    public function handle(Login $event)
```
Comment (deadlybutter) [inline app/Listeners/SuccessfulAuthentication.php]: Do we care about tracking failed logins?

Comment (DFurnes) [inline app/Listeners/SuccessfulAuthentication.php]: We do! Those are tracked by the [`FailedAuthenticationAttempt`](https://github.com/DoSomething/northstar/blob/c910d409a3e537b5ec1d57450316a023d3204164/app/Listeners/FailedAuthenticationAttempt.php) event listener.

**Comment by sheyd** [comment]
🇫🇷  on using `last_authenticated_at`!

**Pull request merged**